### ISSUE TITLE
Respect DesktopEntry (Only)ShowIn

### ISSUE THREAD

**Issue by friday** [Ulauncher/Ulauncher#334]
Some desktop entries should only be shown for specific distros, or hidden from specific distros. Thunar for example depends on the Exo library, and it includes some entries like "Preferred Applications", "Web Browser", "Terminal Emulator" which you don't want to show in other DE's since the have their equivalents.

It should be a very small change to [this condition](https://github.com/Ulauncher/Ulauncher/blob/bf61edc50a79f54cb6934ac81195dc0bd188194f/ulauncher/util/desktop/reader.py#L53) to include [`get_show_in()`](https://developer.gnome.org/gio/stable/gio-Desktop-file-based-GAppInfo.html#g-desktop-app-info-get-show-in).

I would send a PR, but not until the Py3 migration is done.

**Comment by friday** [comment]
Closing in favor of #343 (has more info, and I can only close my own issues anyway)
